geoips/xarray_utils/data.py

```diff
@@ -281,20 +282,43 @@ def sector_xarray_spatial(
 
     if verbose:
         LOG.info("    Padding longitudes")
-    # convert extent longitude to be within 0-360 range
-    min_lon = utils.wrap_longitudes(extent_lonlat[0]) - lon_pad
-    max_lon = utils.wrap_longitudes(extent_lonlat[2]) + lon_pad
-    if min_lon > max_lon and max_lon < 0:
-        max_lon = max_lon + 360
-    min_lat = extent_lonlat[1] - lat_pad
-    max_lat = extent_lonlat[3] + lat_pad
     if verbose:
         LOG.info("    Padding latitudes")
-    # Make sure we don't extend latitudes beyond -90 / +90
-    if min_lat < -90.0:
-        min_lat = -90.0
-    if max_lat > 90.0:
-        max_lat = 90.0
+    extent_lonlat = list(area_def.area_extent_ll) # [min_lon, min_lat, max_lon, max_lat]
+    min_lon, max_lon = -180, 180
+
+    # Check if the area_def includes either North or South Pole. If it does, we have to
+    # manually adjust the lat_lon extent to reflect the correct extent for both lat/lon.
+    # Otherwise go about business as usual.
+    try:
+        LOG.info("Checking if area_def includes the North Pole.")
```

Comment:
Adding this comment so we can prevent merging this branch until v1.12.0-release has been merged.